### PR TITLE
Update adopter.py, allow for other filetypes, add some checking

### PR DESCRIPTION
**Pull request by mazzystr**
Update adopter.py, allow for other filetypes, add some checking

### CODE REVIEW

**Comment by mazzystr** [comment]
fixes #814

File: _data/adopters.py [hunk 1]
```diff
@@ -19,6 +20,21 @@
     }
 }
 
+def logo(name):
+    result = []
+    path = '../assets/images/adopters/'
+    name = re.sub('[^a-zA-Z0-9 \n]', '', name)
+    for root, dirs, files in os.walk(path):
+        result = [s for s in files if name in s]
```
Comment (cwilkers) [inline _data/adopters.py:28]: You could end up with false duplicates, if there were adopters whose names were a subset of another's.

Perhaps, after stripping `name` of unwanted characters, in line 26, you could append a period to do a fake sort of regex `$`?

**Review by cwilkers — COMMENTED** [review]
/lgtm

I'll leave it up to you whether it's worth hacking further to avoid nested company names creating logo conflicts.

File: _data/adopters.py (hunk 1 above)
Comment (mazzystr) [reply _data/adopters.py:28]: > $ touch ../assets/images/adopters/thwarmp.jpg
> $ ./adopters.py
> Curling ADOPTERS file: https://raw.githubusercontent.com/kubevirt/kubevirt/main/ADOPTERS.md
> 
> Image search yielded more than 1 result
> ['arm.svg', 'thwarmp.jpg']
> Please resolve this and settle on one img file

Yup, confirmed.  Brb with a fix.

Comment (mazzystr) [reply _data/adopters.py]: > $ touch ../assets/images/adopters/{armpit,thwarm,thwarmp}.jpg
> $ ./adopters.py
> Curling ADOPTERS file: https://raw.githubusercontent.com/kubevirt/kubevirt/main/ADOPTERS.md
> 
> adopters.yml updated
> Please create pull request... https://github.com/<user>/kubevirt.github.io/pull/new/
> 
> $ cat adopters.yml
> # DO NOT EDIT THIS FILE.  PLEASE RUN ./adopters.py INSTEAD!
> adopters:
>   endusers:
>   - {link: 'https://www.arm.com', logo: arm.svg, name: arm}
>   blah blah blah

This update should fix it.  Good idea on wrapping the string with a delimiter to fake equality in the comparison.  Worked like a charm!

**Review by cwilkers — APPROVED** [review]
/lgtm
/approve

**Pull request merged**